shared/chat/inbox/index.native.js

```diff
@@ -153,7 +158,36 @@ class Inbox extends React.PureComponent<Props, State> {
     this._list = r
   }
 
-  // TODO maybe we can put getItemLayout back if we do a bunch of pre-calc. The offset could be figured out based on index if we're very careful
+  _itemTypeToHeight = {
+    big: globalMargins.large,
+    bigHeader: globalMargins.large,
+    divider: 56,
+    small: 64,
+  }
+
+  // This is an under documented api to help optimize the FlatList's layout. see https://github.com/facebook/react-native/blob/v0.50.0-rc.0/Libraries/Lists/FlatList.js#L118
+  _getItemLayout = (data, index) => {
+    // We cache the divider location so we can divide the list into small and large. We can calculate the small cause they're all
+    // the same height. We iterate over the big since that list is small and we don't know the number of channels easily
```

Comment:
> ...that list is small...

I have 38 total small conversations, and 31 `big` and `bigHeader` rows under my inbox divider. This fcn already assumes that all `small` rows are the same height, we could potentially take advantage of the fact that `_itemTypeToHeight['big'] === _itemTypeToHeight['bigHeader']` to save the iteration at L184. This might be worth doing imo, if I joined all channels in all my teams I'd easily have 2-3x as many big rows as small.

Reply:
cool, we spoke out loud but to add to the documentation. I'm a little worried about taking this out as i can see a world where we have to add it back in when we decide the header should be different heights from the rows ( i don't see this happening for the small rows ). I'm not worried about the perf of running through 100 items and doing some math really. My inbox currently has 5000 small rows so since that math is very easy and i'm confident that'll not change i'd argue to keep that part and leave the iteration of the teams in